### PR TITLE
Add tenant name check in list namespaces function.

### PR DESCRIPTION
**Pull request by zhanghaou**
fix [https://github.com/apache/pulsar/issues/7301](https://github.com/apache/pulsar/issues/7301)

### CODE REVIEW

**Comment by zhanghaou** [comment]
@codelipenghui Done.

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/NamespacesTest.java [hunk 1]
```diff
@@ -252,6 +252,13 @@ public void testGetNamespaces() throws Exception {
         expectedList.sort(null);
         assertEquals(namespaces.getTenantNamespaces(this.testTenant), expectedList);
 
+        try {
+            // check the tenant name is valid
+            namespaces.getTenantNamespaces(this.testTenant + "/default");
+            fail("should have failed");
+        } catch (RestException e) {
```
Comment (codelipenghui) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/NamespacesTest.java:259]: It's better to catch the specific RestException or check the response code.

Comment (zhanghaou) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/NamespacesTest.java:256]: Done.

**Pull request merged**